qiskit/transpiler/target.py

```diff
@@ -985,7 +987,8 @@ def instruction_properties(self, index):
 
     def _build_coupling_graph(self):
         self._coupling_graph = rx.PyDiGraph(multigraph=False)
-        self._coupling_graph.add_nodes_from([{} for _ in range(self.num_qubits)])
+        if self.num_qubits is not None:
+            self._coupling_graph.add_nodes_from([{} for _ in range(self.num_qubits)])
```

Comment:
I added these changes when I was trying to support the edge case with no `coupling_map`. This is no longer necessary to support the edge cases because we are skipping the target, but given that `Target` does technically suport `num_qubits=None`, should we keep these extra checks?